### PR TITLE
Support conv transpos 1D in cuda provider.

### PR DESCRIPTION
**Pull request by zhanghuanrong**
Add ConvTranspose 1D support in cuda provider.
Enable ConvTranspose_1D test for cuda provider.

### CODE REVIEW

File: onnxruntime/core/providers/cuda/nn/conv_transpose.cc [hunk 1]
```diff
@@ -41,14 +41,15 @@ Status ConvTranspose<T>::DoConvTranspose(OpKernelContext* context, bool dynamic_
...
   const TensorShape& x_shape = X->Shape();
-  const auto& x_dims = x_shape.GetDims();
+  auto x_dims = x_shape.GetDims();
   auto x_data = reinterpret_cast<const CudaT*>(X->template Data<T>());
 
-  if (X->Shape().NumDimensions() != 4) {
+  auto x_dimensions = X->Shape().NumDimensions();
+  if (x_dimensions != 4 && x_dimensions != 3) {
     // This condition is not true for two tests in ONNX tests series:
     // test_convtranspose_1d_cpu, test_convtranspose_3d_cpu.
```
Comment (jywu-msft) [inline onnxruntime/core/providers/cuda/nn/conv_transpose.cc]: is the comment about test_convtranspose_1d still valid?

Comment (jywu-msft) [reply onnxruntime/core/providers/cuda/nn/conv_transpose.cc]: Can you also uncomment 
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/test/onnx/main.cc#L576
and see if the test passes now?

Comment (zhanghuanrong) [reply onnxruntime/core/providers/cuda/nn/conv_transpose.cc:50]: Good idea!

Comment (zhanghuanrong) [reply onnxruntime/core/providers/cuda/nn/conv_transpose.cc]: onnx test pass after remove the line.

**Comment by jywu-msft** [comment]
seems like there's a merge conflict which is preventing the CI from running.

**Pull request merged**